### PR TITLE
Refactored the tests to use pytest fixtures

### PR DESCRIPTION
**Pull request by jmagnusson**
Also:
* Add linting and py35 to the test suite
* Running `tox` now tests MySQL, PostgreSQL and SQLite for all python versions
* Document how to pass in DSN string to tox for MySQL and PostgreSQL

### CODE REVIEW

File: tests/test_datetime_defaults.py [hunk 1]
```diff
@@ -1,29 +1,44 @@
...
-                sa.Date, min=date(2000, 1, 1), max=date(2099, 1, 1)
-            )
 
-        self.User = User
+@pytest.mark.usefixtures('lazy_configured', 'Session')
+class TestDateTimeDefaults(object):
 
-    def test_autonow(self):
-        column = self.User.created_at
+    @pytest.mark.skipif(
```
Comment (kvesteri) [inline tests/test_datetime_defaults.py]: Can the test skipping be applied inside method? That way we would be able to fetch to mysql version using SQLAlchemy constructs.

Comment (jmagnusson) [inline tests/test_datetime_defaults.py]: Not using `skipif` AFAIK. Couldn't find anything about it in [its documentation](https://pytest.org/latest/skipping.html).

Comment (kvesteri) [inline tests/test_datetime_defaults.py]: You can use ``pytest.skip``:
```python
def test_something():
    if something:
        pytest.skip('some reason')
```

Comment (jmagnusson) [inline tests/test_datetime_defaults.py]: Ah nice, didn't know about that one. Just pushed a new version.

**Comment by kvesteri** [comment]
Thanks for the PR! I fixed some issues, created [another PR](https://github.com/kvesteri/sqlalchemy-defaults/pull/10) and merged that one :train:

**Comment by jmagnusson** [comment]
Great!

**Pull request closed without merging**